galeracluster/source/streamingreplication.rst

```diff
@@ -0,0 +1,90 @@
+#####################
+Streaming Replication
+#####################
+.. _`sr`:
+.. index::
+   pair: Galera Cluster 4.x; Streaming Replication
+
+Under normal operation, the node performs all replication and certification events when a transaction commits.  When working with small transactions this is fine. However, it poses an issue with long-running writes and changes to large data-sets.
+
+In :term:`Streaming Replication`, the node breaks the transaction into fragments, then certifies and replicates them on the slaves while the transaction is still in progress.  Once certified, the fragment can no longer be aborted by conflicting transactions.
+
+Additionally, Streaming Replication allows the node to process transaction write-sets greater than 2Gb.
+
+.. note:: Streaming Replication is a new feature introduced in version 4.0 of Galera Cluster.  Older versions do not support these operations.
+
+	  
+=================================
+When to Use Streaming Replication
+=================================
+.. _`when-use-sr`:
+
+In most cases, the normal method Galera Cluster uses in replication is sufficient in transferring data from a node to a cluster.  :term:`Streaming Replication` provides you with an alternative for situations in which this is not the case.  Keep in mind that there are some limitations to its use.  It's recommended that you only enable it at a session-level, and then only on specific transactions that require the feature.
+
+.. note:: For more information on the limitations to Streaming Replication, see :ref:`Limitations <sr-limit>`.
+
+
+-------------------------------
+Long-Running Write Transactions
+-------------------------------
+.. _`longrun-write-trx`:
+
+When using normal replication, you may occasionally encounter issues with long-running write transactions.
+
+The longer it takes for a node to commit a transaction, the greater the likelihood that the cluster will apply a smaller, conflicting transaction before the longer one can replicate to the cluster.  When this happens, the cluster aborts the long-running transaction.
+
+Using :term:`Streaming Replication` on long-running transactions mitigates this situation.  Once the node replicates and certifies a fragment, it is no longer possible for other transactions to abort it.
```

Comment:
This is not entirely correct: Certification keys are generated from record locks, therefore they don't cover gap locks or next key locks. If the transaction takes a gap lock, it is possible that a transaction which is executed on another node applies a write set which encounters the gap log and will abort the streaming transaction.

Reply:
This will be addressed in the following revisions.